src/Data/Codec.purs

```diff
@@ -14,66 +14,66 @@ import Data.Profunctor.Star (Star(..))
 import Data.Tuple (Tuple(..))
 
 -- | A general type for codecs.
-data GCodec m n a b = GCodec (m b) (Star n a b)
+data GCodec decoderF encoderF encoderInputType decoderTargetType = GCodec (decoderF decoderTargetType) (Star encoderF encoderInputType decoderTargetType)
```

Comment:
it is called `decoderTargetType` in `(Star encoderF encoderInputType decoderTargetType)`

because `decoderTargetType` is ignored for encoder in `Codec`